snowpack/src/hmr-server-engine.ts

```diff
@@ -34,16 +43,16 @@ export class EsmHmrEngine {
   private currentBatch: HMRMessage[] = [];
   private currentBatchTimeout: NodeJS.Timer | null = null;
   private cachedConnectErrors: Set<HMRMessage> = new Set();
-  wsUrl = `ws://localhost:${DEFAULT_PORT}`;
 
-  constructor(options: {server?: http.Server | http2.Http2Server; delay?: number} = {}) {
-    const wss = options.server
+  constructor(options: EsmHmrEngineOptions) {
+    const wss = 'server' in options 
```

Comment:
I'd rather revert the `in options` checks here and keep the current TS-aware check, and update the TS interface instead:

```
type EsmHmrEngineOptions = ({
  server: http.Server | http2.Http2Server;  
  port?: undefined,
} | {
  port: number;
  server?: undefined;
}) & EsmHmrEngineOptionsCommon;
```

I think that will work, at least